docs/DeveloperGuide.adoc

```diff
@@ -767,14 +912,144 @@ Use case ends.
 +
 Use case resumes at step 2.
 
-{More to be added}
+[discrete]
+=== Use case: Select person
+
+*MSS*
+
+1. User requests to list persons
+2. AddressBook shows a list of persons
+3. User requests to select a specific person in list
+4. AddressBook shows selected person
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 3a. The given index is invalid.
+[none]
+** 3a1. AddressBook shows an error message.
++
+Use case resumes at step 2.
+
+[discrete]
+=== Use case: View history
+
+*MSS*
+
+1. User requests to view history of commands
+2. AddressBook shows a list of history
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. History is empty.
++
+Use case ends.
+
+[discrete]
+=== Use case: Undo
+
+*MSS*
+
+1. User requests to undo commands
+2. AddressBook retrieve previous state of AddressBook
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. No command to undo.
++
+Use case ends.
+
+[discrete]
+=== Use case: Redo
+
+*MSS*
+
+1. User requests to redo commands
+2. AddressBook retrieve previous state of AddressBook
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. No command to redo.
++
+Use case ends.
+
+[discrete]
+=== Use case: Clear
+
+*MSS*
+
+1. User requests to clear persons
+2. AddressBook clear all persons
++
+Use case ends.
+
+[discrete]
+=== Use case: Exit
+
+*MSS*
+
+1. User requests to exit AddressBook
+2. AddressBook closes and exit window
++
+Use case ends.
+
+[discrete]
+=== Use case: Rename Tag
+
+*MSS*
+
+1. User requests to list all existing tags
+2. AddressBook shows a list of tags
+3. User requests to rename a tag in the list
+4. AddressBook renames the tag
+5. AddressBook updates the tag
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. The list is empty.
++
+Use case ends.
+
+* 3a. The given index is invalid.
++
+[none]
+** 3a1. AddressBook shows an error message.
++
+Use case resumes at step 2.
+
+* 3b. Updated name already exist in the list of existing tags
++
+[none]
+** 3b1. AddressBook shows an error message.
++
+Use case resumes at step 2.
 
 [appendix]
 == Non Functional Requirements
 
 .  Should work on any link:#mainstream-os[mainstream OS] as long as it has Java `1.8.0_60` or higher installed.
 .  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
 .  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+.  A user should be able to fully understand and use the AddressBook by following the User Guide and Help UI.
+.  The GUI should be easily readable without straining the user’s eyes.
```

Comment:
`straining the user's eyes` sounds a little too specific. What if the GUI is just uncomfortable for the user (e.g. mixed font sizes). Why not generalise this NFR?